plugins/dell-dock/fu-dell-dock-status.c

```diff
@@ -103,12 +103,20 @@ fu_dell_dock_status_open(FuDevice *device, GError **error)
 	if (fu_device_get_proxy(device) == NULL)
 		fu_device_set_proxy(device, fu_device_get_parent(device));
 
+	if (fu_device_get_proxy(device) == NULL) {
+		g_set_error_literal(error, FWUPD_ERROR, FWUPD_ERROR_INTERNAL, "no proxy");
+		return FALSE;
+	}
 	return fu_device_open(fu_device_get_proxy(device), error);
 }
 
 static gboolean
 fu_dell_dock_status_close(FuDevice *device, GError **error)
 {
+	if (fu_device_get_proxy(device) == NULL) {
+		g_set_error_literal(error, FWUPD_ERROR, FWUPD_ERROR_INTERNAL, "no proxy");
+		return FALSE;
+	}
```

Comment:
likewise, I think this is also impossible.  You wouldn't be able to open if the proxy was NULL.

Reply:
Nah, we close() even if open() failed

Reply:
While changing that plug-in, do you think we could add emulation data now?

Reply:
While changing that plug-in, do you think we could add emulation data now?

Reply:
Yup, I've got a physical printed off list of plugins and I'm working my way through them. It takes fricken ages, but it's 100% worthwhile as it's finding real bugs too.